### PR TITLE
Add version histories info into GetMutableSate response

### PR DESCRIPTION
**Pull request by yux0**
Add version histories info into GetMutableSate response

### CODE REVIEW

File: common/persistence/versionHistory.go [hunk 1]
```diff
@@ -165,3 +167,17 @@ func (h *VersionHistories) AddHistory(item VersionHistoryItem, local VersionHist
 func (h *VersionHistories) GetHistories() []VersionHistory {
 	return h.versionHistories
 }
+
+// ToThrift return thrift format of version histories
+func (h *VersionHistories) ToThrift() *shared.VersionHistories {
+	tHistories := &shared.VersionHistories{}
+	for _, history := range h.versionHistories {
+		tHistory := &shared.VersionHistory{BranchToken: history.branchToken}
```
Comment (wxing1292) [inline common/persistence/versionHistory.go]: i think we should make a copy here

File: idl/github.com/uber/cadence/shared.thrift [hunk 2]
```diff
@@ -1413,3 +1413,20 @@ struct HistoryBranch{
   20: optional string branchID
   30: optional list<HistoryBranchRange>  ancestors
 }
+
+// VersionHistoryItem contains signal eventID and the corresponding version
+struct VersionHistoryItem{
+  10: optional i64 eventID
```
Comment (wxing1292) [inline idl/github.com/uber/cadence/shared.thrift]: `i64 (js.type = "Long") `

File: service/history/mutableState.go [hunk 3]
```diff
@@ -115,6 +115,7 @@ type (
 		GetStartEvent() (*workflow.HistoryEvent, bool)
 		GetContinueAsNew() *persistence.CreateWorkflowExecutionRequest
 		GetCurrentBranch() []byte
+		GetAllBranches() persistence.VersionHistories
```
Comment (wxing1292) [inline service/history/mutableState.go]: GetAllVersionHistory?

File: service/history/mutableStateBuilder.go [hunk 4]
```diff
@@ -81,6 +81,7 @@ type (
 
 		executionInfo    *persistence.WorkflowExecutionInfo // Workflow mutable state info.
 		replicationState *persistence.ReplicationState
+		versionHistories persistence.VersionHistories
```
Comment (wxing1292) [inline service/history/mutableStateBuilder.go]: plz change this to a pointer so we can use `null`

File: common/persistence/versionHistory.go [hunk 5]
```diff
@@ -165,3 +167,17 @@ func (h *VersionHistories) AddHistory(item VersionHistoryItem, local VersionHist
 func (h *VersionHistories) GetHistories() []VersionHistory {
 	return h.versionHistories
 }
+
+// ToThrift return thrift format of version histories
+func (h *VersionHistories) ToThrift() *shared.VersionHistories {
```
Comment (wxing1292) [inline common/persistence/versionHistory.go]: suggestion, also add a function `ToThrift ` on VersionHistory

File: service/history/historyEngine.go [hunk 6]
```diff
@@ -581,6 +581,7 @@ func (e *historyEngineImpl) getMutableState(ctx ctx.Context,
 
 	executionInfo := msBuilder.GetExecutionInfo()
 	execution.RunId = context.getExecution().RunId
+	versionHistories := msBuilder.GetAllBranches()
```
Comment (wxing1292) [inline service/history/historyEngine.go]: could you also add a TODO in GetMutableState function (the one above), specifying that we need to take branching into consideration?

File: common/persistence/versionHistory.go (hunk 1 above)
Comment (yux0) [reply common/persistence/versionHistory.go]: Done

File: idl/github.com/uber/cadence/shared.thrift (hunk 2 above)
Comment (yux0) [reply idl/github.com/uber/cadence/shared.thrift]: Done

File: service/history/mutableStateBuilder.go (hunk 4 above)
Comment (yux0) [reply service/history/mutableStateBuilder.go]: Done

File: common/persistence/versionHistory.go (hunk 5 above)
Comment (yux0) [reply common/persistence/versionHistory.go]: Good point. done

**Pull request merged**